### PR TITLE
Add TypeScript default value mappers

### LINKED ISSUE #6765: [typescript-angular] template variable `defaultValue` not set correctly
##### Description

When overriding the mustache files for `typescript-angular`, I found out it does not set the template variable `defaultValue` for strings. I wanted to file a PR for this, but I noticed the `toDefaultValue` in `DefaultCodegen.java` is also ignoring `StringProperty`s.

##### Swagger-codegen version

2.2.3, 2.3.0-latest

##### Swagger declaration file content or url

N/A

##### Command line used for generation

N/A

##### Steps to reproduce

* create a swagger file with a schema with an property of type 'string' and a 'default'
* update `modules/swagger-codegen/src/main/resources/typescript-angular/modelGeneric.mustache` to contain `defaultValue`
* run the codegen
* the generated .ts file has parses `defaultValue` of `'null'`

##### Related issues/PRs

N/A

##### Suggest a fix/enhancement

Either patching `modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java` or `modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java` by adding

```
if (p instanceof StringProperty) {
    StringProperty sp = (StringProperty) p;
    if (sp.getDefault() != null) {
        return sp.getDefault();
    }
    return "null";
}
```

to `public String toDefaultValue(Property p)`.

I am willing to do so, but I am not sure it can be added to `DefaultCodegen.java` safely.

### PR DESCRIPTION
**Pull request by trenneman**
Fixes #6765.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR

Added a method to set the default values for TypeScript conversions.

### CODE REVIEW

**Comment by wing328** [comment]
@trenneman thanks for the PR.

cc @TiFu  @taxpon  @sebastianhaas @kenisteward @Vrolijkx

**Comment by kenisteward** [comment]
@trenneman Thanks for the PR.  The changes look harmless but can I ask why it was made?  I don't think we use these anywhere (included in the actual typescript client types).  code wide i only see this function being used in 
AbstractJavaCodegen.java
ApexClientCodegen.java
DartClientCodegen.java

which none of these are related to typescript.

**Comment by wing328** [comment]
The CI failure (caused by Ruby client) has been fixed in the latest master already.

**Comment by trenneman** [comment]
@kenisteward For the default mustache templates nothing has changed indeed. However, when providing your own mustache files, it is not possible to use the default values as they are set incorrectly.

I've rebased to master.

**Comment by kenisteward** [comment]
@trenneman Thanks for the clarification.

**Comment by trenneman** [comment]
@kenisteward No problem. It would have made sense to put it in the description.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java [hunk 1]
```diff
@@ -222,6 +231,49 @@ public String getTypeDeclaration(Property p) {
     }
 
     @Override
+    public String toDefaultValue(Property p) {
+        if (p instanceof StringProperty) {
```
Comment (TiFu) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java]: I'm wondering if the code in this method could be cleaner/shorter.

This method treats DoubleProperty, FloatProperty, IntegerProperty (and others) exactly the same. 
I think a better solution might be to do:

```
if (p instanceof Integerproperty 
    || p instanceof FloatProperty
    || p instanceof ....) {
    if (p.getDefault() != null) return p.getDefault().toString()
} else if (p instanceof StringProperty) {
 /* code here */
}
// default if the property doesn't have a proper default (e.g. DateProperty)
return UNDEFINED_VALUE 
```

I'm not set on this, but the current code has a lot of repetitions of the same statements.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java [hunk 2]
```diff
@@ -222,6 +231,49 @@ public String getTypeDeclaration(Property p) {
...
+            StringProperty sp = (StringProperty) p;
+            if (sp.getDefault() != null) {
+                return "\"" + sp.getDefault() + "\"";
+            }
+            return UNDEFINED_VALUE;
+        } else if (p instanceof BooleanProperty) {
+            return UNDEFINED_VALUE;
```
Comment (TiFu) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java]: Wouldn't ```false``` be a better default for a boolean property? (or whatever the default in ```BooleanProperty``` is?)

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java (hunk 1 above)
Comment (trenneman) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java]: I would prefer that as well, but `getDefault()` is not an abstract method of `Property`. Actually, this is the way it is handled in the super class `DefaultCodegen` as well.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java (hunk 2 above)
Comment (trenneman) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java]: In TypeScript, the default is `undefined`, not `false`.

**Comment by trenneman** [comment]
@wing328 The build still fails, can it be re-triggered? I don't think the issue is on my end.

**Comment by wing328** [comment]
@trenneman restarted. MacOS image seems to fail a lot more often so likely we'll switch back to Linux image.

**Comment by trenneman** [comment]
@wing328 OK, thanks!

**Comment by kenjones-cisco** [comment]
After pulling in the latest changes to master, and running `mvn package`, I'm now seeing the following errors:

```
complexPropertyTest(io.swagger.codegen.typescript.typescriptangular2.TypeScriptAngular2ModelTest)  Time elapsed: 0.017 sec  <<< FAILURE!
java.lang.AssertionError: expected [null] but found [undefined]
        at io.swagger.codegen.typescript.typescriptangular2.TypeScriptAngular2ModelTest.complexPropertyTest(TypeScriptAngular2ModelTest.java:124)

listPropertyTest(io.swagger.codegen.typescript.typescriptangular2.TypeScriptAngular2ModelTest)  Time elapsed: 0.003 sec  <<< FAILURE!
java.lang.AssertionError: expected [null] but found [undefined]
        at io.swagger.codegen.typescript.typescriptangular2.TypeScriptAngular2ModelTest.listPropertyTest(TypeScriptAngular2ModelTest.java:91)

simpleModelTest(io.swagger.codegen.typescript.typescriptangular2.TypeScriptAngular2ModelTest)  Time elapsed: 0.003 sec  <<< FAILURE!
java.lang.AssertionError: expected [null] but found [undefined]
        at io.swagger.codegen.typescript.typescriptangular2.TypeScriptAngular2ModelTest.simpleModelTest(TypeScriptAngular2ModelTest.java:45)


Results :

Failed tests:
  TypeScriptAngular2ModelTest.complexPropertyTest:124 expected [null] but found [undefined]
  TypeScriptAngular2ModelTest.listPropertyTest:91 expected [null] but found [undefined]
  TypeScriptAngular2ModelTest.simpleModelTest:45 expected [null] but found [undefined]
```

@wing328

**Comment by kenisteward** [comment]
The tests need to be updated to check for undefined. I don't think they
were

**Comment by kenisteward** [comment]
@trennemen could you update the model tests?

@wing328 are the model tests not run for the ci? If not we need to add them so this doesn't happen again.

**Comment by kenjones-cisco** [comment]
Looks like maybe these were old tests that had been removed or renamed. I ran `mvn clean` and then `mvn package` again and it was successful. 

Hadn't run into this scenario previously, my apologizes.

**Comment by kenisteward** [comment]
Np @tenneman flase alarm haha

**Comment by wing328** [comment]
@kenjones-cisco No worry :)

(The model tests for TS generators (node, angular, etc) are updated as part of this PR)

**Comment by kenisteward** [comment]
K thanks! I didn't realize that

**Pull request merged**